n_request.c

```diff
@@ -283,34 +283,47 @@ J *NoteRequestResponseWithRetry(J *req, uint32_t timeoutSeconds)
  @brief Send a request to the Notecard and return the response.
 
  Unlike NoteRequestResponse, this function expects the request to be a valid
- JSON C-string, rather than a `J` object. It also returns the response as a
- dynamically allocated JSON C-string. The caller is responsible for freeing this
- string.
+ JSON C-string, rather than a `J` object. This string may be newline-terminated,
```

Comment:
Is Notecard happy not receiving the CR-LF, or is it added if not present? (I'm having trouble in github tracing `_Transaction`.). Either way, this sounds like a good candidate for HIL tests to validate that the notecard is happy with both alternatives (or as a unit test here to validate the end of frame mark is added in the code here, if that's the case.)

Reply:
The Notecard is fine with either \r\n or \n. It's weird, but for serial, we always terminate the request to the Notecard with \r\n. For I2C, it's always \n. @zfields will remember better why Ray wanted it this way; I don't know/remember the details.

Reply:
The Windows 7 USB driver was not guaranteed to flush streams that did not end `\r\n`, which is why it exists on serial communication. The Notecard is perfectly content with a solitary `\n` to punctuate a Note.